activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -71,6 +71,8 @@ def parse_as_number(raw_value, precision, scale, option = nil)
           raw_value if raw_value.is_a?(Range)
         elsif raw_value.is_a?(Float)
           parse_float(raw_value, precision, scale)
+        elsif raw_value.is_a?(BigDecimal)
```

Comment:
when I was adding the missing test I've realized that we didn't round `BigDecimal` when scale was given, therefore I've added this.